packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -821,7 +880,7 @@ class IntervalCollectionFactory
         return collection;
     }
 
-    public store(value: IntervalCollection<Interval>): ISerializedInterval[] {
+    public store(value: IntervalCollection<Interval>): ISerializedIntervalV2 {
         return value.serializeInternal();
```

Comment:
rather than compress/decompress within the `serialize` method of `Interval`, I think it would be less invasive to either do it on `load/store` here or in `IntervalCollection.serializeInternal`. Since you currently decompress in `IntervalCollection`'s constructor, I think `IntervalCollection.serializeInternal` would make the most sense. This avoids shifting between formats for most of the places we don't particularly care if the old format is used.